src/core/qgsdistancearea.cpp

```diff
@@ -316,6 +354,14 @@ double QgsDistanceArea::measureLine( const QgsPointXY &p1, const QgsPointXY &p2
 {
   double result;
 
+  if ( willUseEllipsoid() )
+  {
+    if ( !mGeod )
+      computeAreaInit();
+    if ( !mGeod )
+      return 0;
```

Comment:
Does this warrant a warning?

Reply:
@m-kuhn I'm tempted to introduce a new exception for this -- what do you think?

Reply:
Tough question, it's the right thing to do. But...
Do we have a way to know where to catch that? I think it's hard to tell with c++ exceptions?
What do we do with python plugins suddenly starting to except (i.e. is this an API break)? Should we just enable it on debug builds for now and postpone enabling it on release builds only with QGIS 4? That will result in a huge precompiler orgy.

Reply:
I actually doubt it's ever even possible to reach this situation, which is why I'd like an exception (or an assert?)so we know if we DO need to handle it...

Reply:
A `Q_ASSERT` before the `return 0` sounds like the perfect solution to me in this case.

Reply:
Doesn't Q_Assert just crash the program? Or I just had bad experiences with it.

Reply:
It does. But only in debug builds which means we will get a feedback with stack trace but only from testers (and not from production) which is good to ensure situation that are not expected to happen.

Reply:
Ah I didn't know this! Thank you for this clarification. I was always surprised as to why qgis didn't crash more often with all those assert lying about.

Reply:
Ok, asserts added